pkg/api/meta/interfaces.go

```diff
@@ -40,6 +40,8 @@ type Interface interface {
 	SetNamespace(namespace string)
 	Name() string
 	SetName(name string)
+	GenerateName() string
```

Comment:
I think at this point we should be using api.ObjectMetaFor() instead of metadata.Accessor.  Most of the meta abstraction is unnecessary (and much less efficient).